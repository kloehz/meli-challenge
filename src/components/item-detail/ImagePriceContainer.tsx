import useCurrency from "../../hooks/useCurrency";
import { Item } from "../../types/getItemDetail"
import { Prices } from "../prices/Prices";


export const ImagePriceContainer = ({picture, condition, sold_quantity, title, price}: Item) => {
    const parsedPrice = useCurrency({...price});
    return (
        <div className="item-image-price-container">
            <div className="item-detail-image-container">
                <img src={picture}/>
            </div>
            <div className="item-detail-to-buy-container">
                <span className="item-detail-condition-selled" >
                    {condition} - {sold_quantity} vendido(s)
                </span>
                <h1 className="item-detail-title">
                    {title}
                </h1>
                <Prices price={parsedPrice} style='item-detail' />
            <button className="item-detail-buy-button">Comprar</button>
            </div>
        </div>
    )
}
