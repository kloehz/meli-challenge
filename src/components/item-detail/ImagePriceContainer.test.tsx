import { shallow } from "enzyme"
import { ImagePriceContainer } from "./ImagePriceContainer"

const itemMock = {
    "id":"MLA912120163",
        "title":"Set X6 Vasos Vidrio Color Trago Largo 465 Cc Pulsar Bormioli",
        "price":{"currency":"ARS",
        "amount":1499,
        "decimals":0},
        "picture":"http://http2.mlstatic.com/D_625911-MLA45228911173_032021-O.jpg",
        "condition":"Nuevo",
        "free_shipping":false,
        "sold_quantity":50,
        "description":"descripcion"
    }

describe('ImagePriceContainer tests', () => {
    
    const wrapper = shallow(<ImagePriceContainer {...itemMock}/>);

    test('should be match snapshoot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should render the correct image', () => {
        expect(wrapper.find('img').prop('src')).toBe(itemMock.picture);
    });
    
})
