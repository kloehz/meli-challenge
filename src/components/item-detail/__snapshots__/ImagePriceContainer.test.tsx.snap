// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePriceContainer tests should be match snapshoot 1`] = `ShallowWrapper {}`;
